src/main/java/com/accantosystems/stratoss/vnfmdriver/driver/GrantDriver.java

```diff
@@ -73,7 +79,10 @@ public GrantCreationResponse requestGrant(GrantRequest grantRequest) throws Gran
         final HttpEntity<GrantRequest> requestEntity = new HttpEntity<>(grantRequest, headers);
         final ResponseEntity<Grant> responseEntity;
         try {
+            UUID uuid = UUID.randomUUID();
+            LoggingUtils.logEnabledMDC(grantRequest != null ? grantRequest.toString() : null, MessageType.REQUEST, MessageDirection.RECEIVED, uuid.toString(),MediaType.APPLICATION_JSON.toString(), "https",getRequestProtocolMetaData(url) ,grantRequest.getVnfLcmOpOccId());
```

Comment:
last parameter will throw NullpointerException if grantRequest is null.
Please create local variables, initialise with empty string "" if it is null and log it.